code/modules/ventcrawl/ventcrawl_atmospherics.dm

```diff
@@ -0,0 +1,90 @@
+/obj/machinery/atmospherics/var/image/pipe_image
+
+/obj/machinery/atmospherics/Destroy()
+	for(var/mob/living/M in src) //ventcrawling is serious business
+		M.remove_ventcrawl()
+		M.forceMove(get_turf(src))
+	if(pipe_image)
+		for(var/mob/living/M in player_list)
+			if(M.client)
+				M.client.images -= pipe_image
+				M.pipes_shown -= pipe_image
+		pipe_image = null
+	. = ..()
+
+/obj/machinery/atmospherics/ex_act(severity)
+	for(var/atom/movable/A in src) //ventcrawling is serious business
+		A.ex_act(severity)
+	. = ..()
+
+/obj/machinery/atmospherics/Entered(atom/movable/Obj)
+	if(istype(Obj, /mob/living))
+		var/mob/living/L = Obj
+		L.ventcrawl_layer = layer
+	. = ..()
+
+/obj/machinery/atmospherics/relaymove(mob/living/user, direction)
+	if(!(direction & initialize_directions)) //can't go in a way we aren't connecting to
+		return
+	ventcrawl_to(user,findConnecting(direction, user.ventcrawl_layer),direction)
+
+/obj/machinery/atmospherics/proc/ventcrawl_to(var/mob/living/user, var/obj/machinery/atmospherics/target_move, var/direction)
+	if(target_move)
+		if(is_type_in_list(target_move, ventcrawl_machinery) && target_move.can_crawl_through())
+			if(istype(target_move, /obj/machinery/atmospherics/unary/vent_pump/) && target_move:is_welded())
+				usr.visible_message("<span class='warning'You hear something banging on \the [target_move.name]!</span>", "<span class='notice'>You can't escape from a welded vent.</span>")
+			else
+				user.remove_ventcrawl()
+				user.forceMove(target_move.loc) //handles entering and so on
+				user.sight &= ~(SEE_TURFS|BLIND)
+				usr.visible_message("<span class='warning'>You hear something squeezing through the ducts.</span>", "You climb out the ventilation system.")
```

Comment:
This should use `user` instead of `usr`. Same for every other use of `usr` in this proc.